misk-redis/src/main/kotlin/misk/redis/RealRedis.kt

```diff
@@ -302,6 +303,51 @@ class RealRedis(
     jedisPool.resource.use { jedis -> jedis.publish(channel, message) }
   }
 
+  override fun zadd(key: String, score: Double, member: ByteString, options: Set<String>): Long {
+    return jedisPool.resource.use { jedis ->
+      jedis.zadd(key.toByteArray(charset), score, member.toByteArray(), getZAddParams(options))
+    }
+  }
+
+  private fun getZAddParams(options: Set<String>): ZAddParams {
+    val params = ZAddParams()
+
+    options.forEach {
+      when(ZAddOptions.valueOf(it.uppercase())) {
```

Comment:
Should we add some sort validation before this? otherwise this will throw an exception.

Reply:
> Should we add some sort validation before this?

Won't the validation step throw an exception too? Or did you mean to filter out invalid options? I was intending this step to throw an exception if an invalid value was provided.

Reply:
Maybe if we do it like you suggested above Set<ZAddOptions> that might eliminate the need to do this conversation/validation at all.